src/storage/sdr/record/mod.rs

```diff
@@ -627,24 +622,30 @@ impl<'a> TypeLengthRaw<'a> {
     }
 }
 
-impl<'a> From<TypeLengthRaw<'a>> for SensorId {
-    fn from(value: TypeLengthRaw<'a>) -> Self {
+impl<'a> TryFrom<TypeLengthRaw<'a>> for SensorId {
+    type Error = ParseError;
+
+    fn try_from(value: TypeLengthRaw<'a>) -> Result<Self, Self::Error> {
         let TypeLengthRaw(value, data) = value;
         let type_code = (value >> 6) & 0x3;
 
         let length = value & 0x1F;
 
         let data = &data[..(length as usize).min(data.len())];
 
-        let str = core::str::from_utf8(data).map(ToString::to_string);
-
-        match type_code {
-            0b00 => SensorId::Unicode(str.unwrap()),
-            0b01 => SensorId::BCDPlus(data.to_vec()),
-            0b10 => SensorId::Ascii6BPacked(data.to_vec()),
-            0b11 => SensorId::Ascii8BAndLatin1(str.unwrap()),
+        let id = match type_code {
+            0b00 => SensorId::Unicode(
+                core::str::from_utf8(data)
+                    .map_err(|_| ParseError::InvalidSensorId)
+                    .map(str::to_string)?,
+            ),
+            0b01 => todo!(),
+            0b10 => todo!(),
```

Comment:
why leave these as todo's? 

We've never seen anything other than unicode, so I'd be okay with turning these into ParseError's, but I'd prefer these not panic.

Reply:
Good point, wasn't thinking about this super hard. Replaced them with what we had before, and fixed up the display function to just print somewhat bogus data. We can correct it some other time.